src/gl/canvas2dRe.re

```diff
@@ -0,0 +1,76 @@
+type t;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : unit = "" [@@bs.send.pipe: t];
+external restore : unit = "" [@@bs.send.pipe: t];
+
+/* Transformation */
+external scale : float => float => unit = "" [@@bs.send.pipe: t];
+external rotate : float => unit = "" [@@bs.send.pipe: t];
+external translate : float => float => unit = "" [@@bs.send.pipe: t];
+external transform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+external setTransform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+
+/* Compositing */
+external globalAlpha : t => float => unit = "" [@@bs.set];
+external globalCompositeOperation : t => string => unit = "" [@@bs.set];
```

Comment:
Thanks @glennsl, `type t = private string` is actually the first thing I tried (inside the sub-module definitions), but the compiler gives me `Unbound type constructor private`. AFAICT this is valid OCaml, but maybe Reason has different syntax for this?

The `Composite` module is also fixed now (a silly oversight, sorry!). If you're happy with it as-is, and I can't figure out the private type declaration (which admittedly would be a lot nicer than the verbose module types), it'd be nice to have this merged and then I can iterate in smaller PRs as needed.